src/tokens/tokens.service.ts

```diff
@@ -141,13 +149,26 @@ export class TokensService {
     );
 
     this.supportsCustomUri = await this.queryUriSupport();
+
+    if (!this.contractAddress) {
```

Comment:
`init()` is not guaranteed to be called on every start (only has to be called once ever). So we need to be sure `this.contractAddress` is initialized somewhere else. Might want to do it lazily the first time a query/transaction happens?

Reply:
Come to think of it, there's probably an existing bug with the initialization of `this.supportsCustomUri` above as well.